megablocks/layers/gelu.py

```diff
@@ -1,7 +1,72 @@
 import stk
+import torch
 import torch.nn.functional as F
 
-def gelu(x):
+
+@torch.jit.script
+def _gelu_forward(x):
+    return  x * 0.5 * (1.0 + torch.tanh(0.79788456 * x * (1 + 0.044715 * x * x)))
+
+
+@torch.jit.script
+def _gelu_forward_inplace(x):
+    return  x.mul_(0.5 * (1.0 + torch.tanh(0.79788456 * x * (1 + 0.044715 * x * x))))
+
+
+@torch.jit.script
+def _gelu_backward_inplace(g, x):
+    tanh_out = torch.tanh(0.79788456 * x * (1 + 0.044715 * x * x))
+    ff = (
+        0.5 * x * (
+            (1 - tanh_out * tanh_out) *
+            (0.79788456 + 0.1070322243 * x * x)
+        ) + 0.5 * (1 + tanh_out)
+    )
+    return g.mul_(ff)
+
+
+def gelu_forward(x: stk.Matrix):
+    assert isinstance(x, stk.Matrix)
+    return stk.Matrix(
+        x.size(),
+        _gelu_forward(x.data),
+        x.row_indices,
+        x.column_indices,
+        x.offsets,
+        x.column_indices_t,
+        x.offsets_t,
+        x.block_offsets_t)
+
+
+def gelu_forward_inplace(x: stk.Matrix):
```

Comment:
A note on convention
in pytorch, inplace ops generally just have an `_` at the end of the name
ie instead of `gelu_forward_inplace` I'd just call it `gelu_forward_` to follow convention.

eg [mul_](https://pytorch.org/docs/stable/generated/torch.Tensor.mul_.html)

Reply:
Done.